src/OrbitGl/TranslationStack.h

```diff
@@ -0,0 +1,27 @@
+// Copyright (c) 2021 The Orbit Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#ifndef ORBIT_GL_TRANSFORMATION_STACK_H_
+#define ORBIT_GL_TRANSFORMATION_STACK_H_
+
+#include <vector>
+
+#include "CoreMath.h"
+
+namespace orbit_gl {
+class TranslationStack {
+ public:
+  void PushTranslation(float x, float y, float z = 0.f);
+  void PopTranslation();
+  [[nodiscard]] bool IsEmpty() { return translation_stack_.empty(); }
+
+  [[nodiscard]] Vec3 TransformAndFloorVertex(const Vec3& input) const;
```

Comment:
Same as above (Transform/Translate). I would stick to one term, otherwise I as a user asking myself if that's a different operation.